main.go

```diff
@@ -83,11 +87,24 @@ func run(ctx context.Context) {
 			return
 		}
 
-		// This sets up the patching system to run in the background.
-		ospatchog.Configure(ctx)
+		if config.TaskNotificationEnabled() && (taskNotificationClient == nil || taskNotificationClient.Closed()) {
+			// Start WaitForTaskNotification if we need to.
+			taskNotificationClient, err = agentendpoint.NewClient(ctx)
+			if err != nil {
+				logger.Errorf(err.Error())
```

Comment:
Under what conditions would this return nil and is it ok to continue after this error is received? Will we retry to get the client?

Reply:
We don't want the entire service to fail if client dialing does (that should be very rare as the dialing happens async, otherwise its a configuration error on our part), we will simple retry in 10m

If this ends up being an error we actually see we can add addition shorter retry logic.